controllers/openstackdataplane_controller.go

```diff
@@ -205,12 +205,21 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 			err.Error()))
 		return ctrl.Result{}, err
 	}
+	if len(roles.Items) > 0 && len(instance.Spec.Roles) > len(roles.Items) {
+		r.Log.Info("found roles", "total", len(roles.Items))
+		shouldRequeue = true
+	}
 	if shouldRequeue {
 		r.Log.Info("one or more roles aren't ready, requeueing")
 		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
 	}
 	if instance.Spec.DeployStrategy.Deploy && len(deployErrors) == 0 {
 		instance.Status.Deployed = true
+		if len(roles.Items) == 0 {
```

Comment:
Isn't dataplane created without any roles a valid scenario? If that's  not true, then we should make `roles` required in dataplane CRD. Here we won't set DeploymentReadyCondition for that scenario.